### PR TITLE
ratelimit: support response_headers_to_add

### PR DESCRIPTION
**Pull request by zirain**
Commit Message: ratelimit: support response_headers_to_add
Additional Description:
  ratelimit support response_headers_to_add like local_ratelimit
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM please add a release note, thank you.

/wait

File: changelogs/current.yaml [hunk 1]
```diff
@@ -86,5 +86,8 @@ new_features:
 - area: access_log
   change: |
     updated command operator ``%GRPC_STATUS%`` to suppoprt the snake case.
+- area: ratelimit
+  change: |
+    add support for :ref:`API version <envoy_v3_api_field_extensions.filters.http.ratelimit.v3.RateLimit.response_headers_to_add>` .
```
Comment (mattklein123) [inline changelogs/current.yaml:91]:
```suggestion
    add support for :ref:`adding response headers <envoy_v3_api_field_extensions.filters.http.ratelimit.v3.RateLimit.response_headers_to_add>` to rate-limited responses.
```

**Review by mattklein123 — CHANGES_REQUESTED** [review]
One final release note nit, thank you.

/wait

File: changelogs/current.yaml (hunk 1 above)
Comment (zirain) [reply changelogs/current.yaml]: fixed

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**